cmd/pow/cmd/prepare/prepare.go

```diff
@@ -395,10 +454,44 @@ func dagify(ctx context.Context, dagService ipld.DAGService, path string, quiet
 			c.Message("DAG created in %.02fs.", time.Since(start).Seconds())
 		}()
 	}
-	dataCid, err := dataprep.Dagify(ctx, dagService, path, progressChan)
-	if err != nil {
-		return cid.Undef, fmt.Errorf("creating dag for data: %s", err)
+
+	var aggregatedFiles []aggregatedFile
+	var dataCid cid.Cid
+	var err error
+	if !aggregate {
+		dataCid, err = dataprep.Dagify(ctx, dagService, path, progressChan)
+		if err != nil {
+			return cid.Undef, nil, fmt.Errorf("creating dag for data: %s", err)
+		}
+	} else {
+		var lst []aggregator.AggregateDagEntry
+		err = filepath.Walk(path, func(p string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+			dcid, err := dataprep.Dagify(ctx, dagService, p, progressChan)
+			if err != nil {
+				return err
+			}
+			lst = append(lst, aggregator.AggregateDagEntry{
+				RootCid: dcid,
+			})
+			aggregatedFiles = append(aggregatedFiles, aggregatedFile{Name: p, Cid: dcid.String()})
+			return err
+		})
+		dataCid, err = aggregator.Aggregate(ctx, dagService, lst)
+		if err != nil {
+			return cid.Undef, nil, fmt.Errorf("aggregating: %s", err)
+		}
 	}
```

Comment:
If `--aggregate` is provided, we walk the folder, ingest each file, and later aggregate it.